### ISSUE TITLE
[Helm] Thirdparty Helm Charts are not scalable due to the Kubernetes Service being a headless service

### ISSUE THREAD

**Issue by mdebarros** [mojaloop/project#3163]
**Summary**:

Thirdparty Helm Charts are not scalable due to the Kubernetes Service being a [headless service](https://kubernetes.io/docs/concepts/services-networking/service/#headless-services), which means that Load Balancing is not possible if the following components are scaled-out:

    1. `tp-api-svc`
    2. `auth-svc`
    3. `consent-oracle`

**Severity**:
Medium

**Priority**:
Low

**Expected Behavior**

Thirdparty Helm Charts should be deployed with a NON [headless service](https://kubernetes.io/docs/concepts/services-networking/service/#headless-services) for the following components:

    1. `tp-api-svc`
    2. `auth-svc`
    3. `consent-oracle`

**Steps to Reproduce**
  1. Deploy Mojaloop v14.1.1
  2. Execute `kubectl -n $SERVICE_NS get svc/$SERVICE_NAME -o json` for each of the above components
  3. Take note that the `clusterIP=none`, and that no clusterIp has been assigned to the service

**Specifications**
  - Component (if known): Helm
  - Version: v14.1.1
  - Platform:  Kubernetes
  - Subsystem: Helm
  - Type of testing: n/a
  - Bug found/raised by: @mdebarros 

**Notes**:

- Severity when opened: Medium
- Priority when opened: Low

**Comment by mdebarros** [comment]
# Work-around

To fix this going forward one must manually re-create (delete & create) the affected services prior to the upgrade by removing the following JSON keys:

- `.spec.clusterIP`
- `.spec.clusterIPs`

*NOTE: Modifying the existing Kubernetes definition will NOT take effect, thus the need to completely re-create the Kubernetes service by deleting it and then creating it!*

Alternatively, you can execute the following commands in your terminal or save it to a bash script, and ensure you update the exported env variables prior to execution:

- `SERVICE_NS` - Namespace of the deployment
- `HELM_NAME` - Helm installation name
- `SERVICE_NAME` - Name of the Service to be re-created. Remove the `$HELM_NAME-` prefix if not required.

Pre-requisites:

- [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl) - Kubernetes command-line tool, with configured access to your target Kubernetes cluster.
- [jq](https://stedolan.github.io/jq/) - A lightweight and flexible command-line JSON processor.

Script to execute:

```bash
export SERVICE_NS="<NAMESPACE>";
export HELM_NAME="<HELM INSTALL NAME>";

export SERVICE_NAME="$HELM_NAME-tp-api-svc"; \
echo "Re-creating the $SERVICE_NS/$SERVICE_NAME for Helm install $HELM_NAME"; \
export K8_DESCRIPTOR=$(kubectl -n $SERVICE_NS get svc/$SERVICE_NAME -o json | jq 'del(.spec.clusterIP) | del(.spec.clusterIPs)') && \
kubectl -n $SERVICE_NS delete svc/$SERVICE_NAME && \
echo $K8_DESCRIPTOR | kubectl create --save-config -f -;

export SERVICE_NAME="$HELM_NAME-auth-svc"; \
echo "Re-creating the $SERVICE_NS/$SERVICE_NAME for Helm install $HELM_NAME"; \
export K8_DESCRIPTOR=$(kubectl -n $SERVICE_NS get svc/$SERVICE_NAME -o json | jq 'del(.spec.clusterIP) | del(.spec.clusterIPs)') && \
kubectl -n $SERVICE_NS delete svc/$SERVICE_NAME && \
echo $K8_DESCRIPTOR | kubectl create --save-config -f -;

export SERVICE_NAME="$HELM_NAME-consent-oracle"; \
echo "Re-creating the $SERVICE_NS/$SERVICE_NAME for Helm install $HELM_NAME"; \
export K8_DESCRIPTOR=$(kubectl -n $SERVICE_NS get svc/$SERVICE_NAME -o json | jq 'del(.spec.clusterIP) | del(.spec.clusterIPs)') && \
kubectl -n $SERVICE_NS delete svc/$SERVICE_NAME && \
echo $K8_DESCRIPTOR | kubectl create --save-config -f -;
```

**Comment by mdebarros** [comment]
This has been resolved in [Mojaloop Helm v15.0.0](https://github.com/mojaloop/helm/pull/551).

**Comment by mdebarros** [comment]
# Update

This has been resolved in [Mojaloop Helm v15.0.0](https://github.com/mojaloop/helm/pull/551).

This has NOT been resolved in Mojaloop Helm v14.x release. Refer to the above [Work-around](https://github.com/mojaloop/project/issues/3163#issuecomment-1469551547) to see how this can be fixed.

**Comment by elnyry-sam-k** [comment]
Decided to add as "known issue" for v14.1.1 and this is fixed in v15.0.0
